politeiad/backendv2/tstorebe/plugins/pi/hooks.go

```diff
@@ -390,11 +392,264 @@ func (p *piPlugin) voteSummary(token []byte) (*ticketvote.SummaryReply, error) {
 	return &sr, nil
 }
 
+// comments requests all comments on a record from the comments plugin.
+func (p *piPlugin) comments(token []byte) (*comments.GetAllReply, error) {
+	reply, err := p.backend.PluginRead(token, comments.PluginID,
+		comments.CmdGetAll, "")
+	if err != nil {
+		return nil, err
+	}
+	var gar comments.GetAllReply
+	err = json.Unmarshal([]byte(reply), &gar)
+	if err != nil {
+		return nil, err
+	}
+	return &gar, nil
+}
+
+// isInCommentTree returns whether the leafID is part of the provided comment
+// tree. A leaf is considered to be part of the tree if the leaf is a child of
+// the root or the leaf references the root itself.
+func isInCommentTree(rootID, leafID uint32, cs []comments.Comment) bool {
+	if leafID == rootID {
+		return true
+	}
+	// Convert comments slice to a map
+	commentsMap := make(map[uint32]comments.Comment, len(cs))
+	for _, c := range cs {
+		commentsMap[c.CommentID] = c
+	}
+
+	// Start with the provided comment leaf and traverse the comment tree up
+	// until either the provided root ID is found or we reach the tree head. The
+	// tree head will have a comment ID of 0.
+	current := commentsMap[leafID]
+	for current.ParentID != 0 {
+		// Check if next parent in the tree is the rootID.
+		if current.ParentID == rootID {
+			return true
+		}
+		leafID = current.ParentID
+		current = commentsMap[leafID]
+	}
+	return false
+}
+
+// latestAuthorUpdate gets the latest author update on a record, if
+// the record has no author update it returns nil.
+func latestAuthorUpdate(token []byte, cs []comments.Comment) (*comments.Comment, error) {
```

Comment:
There are no error paths in this function so there is no need to return an error.

Reply:
👀